frc_characterization/logger_analyzer/data_analyzer.py

```diff
@@ -233,106 +215,28 @@ def runAnalysis():
                 Tests.ARM: runAnalysisArm,
                 Tests.SIMPLE_MOTOR: runAnalysisSimple,
             }
+
+            self.prepared_data = self.prepare_data(
```

Comment:
Apparently numpy dislikes using None in a tuple of np arrays so I had to make an is_valid method that iterates through the tuple to make sure none of the elements are null.

Also since the subsets need to be stored, I decided to just save it to a self.prepared_data dict.